### PR TITLE
Added Chile ID (RUT) validator

### LINKED ISSUE #79: [v1.3.0] Implement Chile ID 🇨🇱 (RUT) validator
- Implement the Chile ID validator based on the [description](https://es.wikipedia.org/wiki/Rol_%C3%9Anico_Tributario) and [algorithm](https://es.wikipedia.org/wiki/Anexo:Implementaciones_para_algoritmo_de_rut) available on Wikipedia
- Write tests
@JoaoFSCruz Would it be possible to relate the issue to the hacktoberfest label? xD

### PR DESCRIPTION
**Pull request by gheleri**
resolve #79
It includes the validator itself and unit tests. The algorithm was based on wikipedia info.

### CODE REVIEW

**Review by AlexOlival — CHANGES_REQUESTED** [review]
Looking great here as well! Just a minor alphabetic order issue. Also, please target the v1.2.0 branch.

Thank you!

**Comment by AlexOlival** [comment]
You'll have to fix the conflicts. We're now namespacing by continent, so make sure to get those right!

**Comment by gheleri** [comment]
Should I namespace as Latin America or Central/South America? Latin America seems easiest

**Comment by gheleri** [comment]
> You'll have to fix the conflicts. We're now namespacing by continent, so make sure to get those right!

Should I namespace as Latin America or Central/South America? Latin America seems easiest

**Comment by gheleri** [comment]
Please check now

**Comment by AlexOlival** [comment]
I think Geography makes more sense here, so SouthAmerica 👍

**Comment by gheleri** [comment]
It's done, mate!

**Review by AlexOlival — APPROVED** [review]

**Comment by AlexOlival** [comment]
Looking good on my end :+1: 
Just needs @JoaoFSCruz to take a look

File: src/Core/SouthAmerica/Brazil/BrazilIdValidator.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        $cpfArray = array_reverse(
+            array_map(
+                static function ($digit) {
+                    return (int) $digit;
+                },
+                str_split($id)
+            )
+        );
+
+        // source: https://pt.wikipedia.org/wiki/Cadastro_de_pessoas_f%C3%ADsicas#Algoritmo
```
Comment (JoaoFSCruz) [inline src/Core/SouthAmerica/Brazil/BrazilIdValidator.php:26]: Although it's not done in every country implementation (my bad 😞), add a docblock for the class to include the source of the algorithm (here is an [example](https://github.com/reducktion/socrates/blob/master/src/Core/Portugal/PortugalIdValidator.php)).

File: src/Core/SouthAmerica/Chile/ChileIdValidator.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+use Illuminate\Support\Str;
+use Reducktion\Socrates\Contracts\IdValidator;
+use Reducktion\Socrates\Exceptions\InvalidLengthException;
+
+class ChileIdValidator implements IdValidator
+{
+    public function validate(string $id): bool
+    {
+        // source: https://es.wikipedia.org/wiki/Anexo:Implementaciones_para_algoritmo_de_rut
```
Comment (JoaoFSCruz) [inline src/Core/SouthAmerica/Chile/ChileIdValidator.php:13]: Ditto as above.

**Review by JoaoFSCruz — CHANGES_REQUESTED** [review]
Wonderful job!!! Just two small changes regarding the sources. Thank you for your contribution 🙏

**Comment by AlexOlival** [comment]
I have fixed the branch name because it was erroneously targetting a tag.
Please change the merge target to release-1.3.0

File: src/Core/SouthAmerica/Brazil/BrazilIdValidator.php (hunk 1 above)
Comment (AlexOlival) [reply src/Core/SouthAmerica/Brazil/BrazilIdValidator.php:26]: This isn't scoped to him, but if he can go ahead and change this after targetting the new `release-1.3.0` branch, we'd be much appreciated :pray:

**Comment by gheleri** [comment]
Done. I already changed the target to release-1.3.0 and add the codeblocks for Brazil, Chile and others minor fixes. However I did not add the code block for other European countries that do not yet have that (I realized it is necessary to add it when there are comments)

**Review by JoaoFSCruz — APPROVED** [review]
Great! Don't worry about the other docblocks. It's a long long task to be undertaken. As we go we'll add the sources and anything that's needed.

Thank you so much for your contribution 🙏

**Comment by AlexOlival** [comment]
Valeu @gheleri! 😁
@all-contributors please add @gheleri for code and documentation

**Pull request merged**